Changed Backbone.couch_connector.update to use $.couch.db.updateDoc

gijohnson-usgs opened the pull request:
Referencing Issue [#42](https://github.com/janmonschke/backbone-couchdb/issues/42), this change allows the use of CouchDB's update handlers.

Usage:

```
con.update(model, {
  updateFun : "update_handler",
  attr1 : attr1,
  attr2 : attr2,
  ...
  success : function(doc) {
    ...
  },
  error : function(doc) {
    ...
  }
})
```

I'm also interested in having Backbone.couch_connector.create call this function if the user provides an "updateFun" option.

gijohnson-usgs committed:
added update function to Backbone.couch_connector

gijohnson-usgs committed:
reformatted update function in backbone-couchdb.coffee

gijohnson-usgs committed:
reformatted backbone-couchdb.coffee w/ 2 space, soft tabs

janmonschke commented:
Looks good to me, but can you merge in latest master first and then run the tests? (https://github.com/janmonschke/backbone-couchdb/blob/master/test/readme.md)

gijohnson-usgs committed:
Checking to see if call to update has a updateFun attr. If not, calling create.

gijohnson-usgs committed:
Merged branch 'master' of https://github.com/janmonschke/backbone-couchdb

gijohnson-usgs commented:
Tests are passing, but I'd like to add some more.

gijohnson-usgs committed:
Add tests for Backbone.couch_connector.update

- Add tests for using the update function w/ an update handler.

- Update test/testrunner/src/jquery.couch.js to
  https://github.com/daleharvey/jquery.couch.js

- Move updateFun from options to the model.

- Allow user to update all model attributes or just ones passed to
  model.save

gijohnson-usgs commented:
I've add more tests and moved the updateFun from opts to the model. I think this is more intuitive, but if you have other suggestions I'd love to hear them.

gijohnson-usgs commented:
Another option is to add `updateDoc` if it doesn't exist.

janmonschke commented:
That's another option, we could do that.
But then we'd also have to maintain that function.

Let's try with including the function.

gijohnson-usgs commented:
Would you like it added to `$.couch.db` or perhaps in `Backbone.couch_connector.helpers`?

gijohnson-usgs committed:
Adding check for $.couch.db.updateDoc

gijohnson-usgs commented:
I decided not to include it b/c `$.couch` is using a few nested closures and `updateDoc` relies on one 'private' method called `encodeOptions`.

gijohnson-usgs committed:
Adding support for versions of jquery.couch.js that don't have updateDoc

gijohnson-usgs committed:
Moving check for updateDoc from make_db to update

gijohnson-usgs commented:
Changed my mind. Now there is a check in `update` that binds `update_doc` and it's dependency `encode_options` to the db object if `$.couch` doesn't contain it. All tests are passing.

gijohnson-usgs committed:
Adding test for update

gijohnson-usgs committed:
Merge branch 'master' of https://github.com/janmonschke/backbone-couchdb

Conflicts:
	backbone-couchdb.js

gijohnson-usgs committed:
Merge remote-tracking branch 'janmonschke/master'

Conflicts:
	backbone-couchdb.js

gijohnson-usgs committed:
Revert "Adding test for update"

This reverts commit 02672759a641f12754dcf4b291d99afc3108075e.

Conflicts:

	backbone-couchdb.coffee
	backbone-couchdb.js

gijohnson-usgs committed:
Reverting in order to rebase.
